drizzlepac/hlautils/astroquery_utils.py

```diff
@@ -44,65 +46,77 @@ def retrieve_observation(obsid, suffix=['FLC'], archive=False,clobber=False):
     """
     local_files = []
 
-    # Query MAST for the data with an observation type of either "science" or "calibration"
-    obsTable = Observations.query_criteria(obs_id=obsid, obstype='all')
+    # Query MAST for the data with an observation type of either "science" or
+    # "calibration"
+    obs_table = Observations.query_criteria(obs_id=obsid, obstype='all')
     # Catch the case where no files are found for download
-    if len(obsTable) == 0:
+    if obs_table:
         log.info("WARNING: Query for {} returned NO RESULTS!".format(obsid))
         return local_files
 
-    dpobs = Observations.get_product_list(obsTable)
-    dataProductsByID = Observations.filter_products(dpobs,
-                                              productSubGroupDescription=suffix,
-                                              extension='fits',
-                                              mrp_only=False)
-
-    # After the filtering has been done, ensure there is still data in the table for download.
-    # If the table is empty, look for FLT images in lieu of FLC images. Only want one
-    # or the other (not both!), so just do the filtering again.
-    if len(dataProductsByID) == 0:
-        log.info("WARNING: No FLC files found for {} - will look for FLT files instead.".format(obsid))
+    dpobs = Observations.get_product_list(obs_table)
+    data_products_by_id = \
```

Comment:
Instead of continuing the line like this, more often you will see the following approach:
```python
data_products_by_id = Observations.filter_products(
    dpobs,
    productSubGroupDescription=suffix,
    extension='fits',
    mrp_only=False
)
```